src/historywork/GetRemoteFileWork.cpp

```diff
@@ -37,13 +41,23 @@ GetRemoteFileWork::getCommand()
     releaseAssert(mCurrentArchive);
     releaseAssert(mCurrentArchive->hasGetCmd());
     auto cmdLine = mCurrentArchive->getFileCmd(mRemote, mLocal);
-
+    // getCommand is called from RunCommandWork::onRun immediately before
+    // running the command.  Record the time here so that we can calculate
+    // throughput in onSuccess.
+    mDownloadStart = std::chrono::steady_clock::now();
+    if (getRetryCount() > 0)
```

Comment:
I don't think this currently works, and the retry metric will always be 0. It looks like in `src/catchup/CatchupWork.cpp`, you set `GetHistoryArchiveStateWork` to retry `HISTORY_ARCHIVE_RETRY_LIMIT` times. `GetHistoryArchiveStateWork` then spawns `GetRemoteFileWork`. The issue is, `GetHistoryArchiveStateWork` spawns `GetRemoteFileWork` with `RETRY_NEVER` at `src/historywork/GetHistoryArchiveStateWork.cpp:87`. What will happen is that `GetHistoryArchiveStateWork` will keep retrying, but on each retry spin up a fresh `GetRemoteFileWork` work. This means that on each iteration, `GetRemoteFileWork.getRetryCount()` will always be 0.

I could be wrong though, since `GetRemoteFileWork` is a child of `GetHistoryArchiveStateWork` it may inherit it's retry counter or something. One way to test this behavior is to run a watcher node, but set invalid URLs for the history archives. You can then run the node and use the `--metric` flag to print your new metrics to the console. This will help you verify that the metric counter is working properly, since it should have to retry 10 times since the URL is invalid.

Reply:
agreed with @SirTyson. You could probably create a new Work wrapper class `DownloadWorkWithMetrics` or something like that, which `GetHistoryArchiveStateWork` and `GetAndUnzipRemoteFileWork` would inherit from. This way we'll automatically have metrics for all archive download works. In that new work, `onFailureRetry` is a good place to update the retry metric, this way you don't need to track retries manually.